prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -928,4 +935,361 @@ class FhirTransformerTests {
+                    )
+                )
+            )
+
+        val transformer = FhirTransformer(schema)
+        transformer.process(bundle)
+        assertThat(serviceRequest.note).hasSize(3)
+        assertThat(serviceRequest.orderDetail).hasSize(4)
+        assertThat(patient.name).hasSize(2)
+    }
+
+    @Test
+    fun `test move Observation to ServiceRequest note`() {
+        val bundle = Bundle()
+        bundle.id = "abc123"
+        val serviceRequest = ServiceRequest()
+        val serviceRequestEntry = bundle.addEntry()
+        serviceRequestEntry.fullUrl = serviceRequest.id
+        serviceRequestEntry.resource = serviceRequest
+        val observation1 = Observation()
+        observation1.id = "obs123"
+        observation1.code.coding.add(
+            Coding(
+                "SNOMED",
+                "92142-9",
+                "Influenza virus A RNA [Presence] in Respiratory specimen by NAA with probe detection"
+            )
+        )
+        observation1.interpretation = mutableListOf(CodeableConcept(Coding("loinc", "260385009", "Negative")))
+        val observationEntry1 = bundle.addEntry()
+        observationEntry1.fullUrl = observation1.id
+        observationEntry1.resource = observation1
+        val observation2 = Observation()
+        observation2.id = "obs123"
+        observation2.code.coding.add(
+            Coding(
+                "SNOMED",
+                "92141-1",
+                " Influenza virus B RNA [Presence] in Respiratory specimen by NAA with probe detection"
+            )
+        )
+        observation2.interpretation = mutableListOf(CodeableConcept(Coding("loinc", "260385009", "Negative")))
+        val observationEntry2 = bundle.addEntry()
+        observationEntry2.fullUrl = observation2.id
+        observationEntry2.resource = observation2
+
+        val schema = ConfigSchemaReader.fromFile(
+            "classpath:/fhir_sender_transforms/convert-all-obs-to-note.yml",
+            schemaClass = FhirTransformSchema::class.java,
+            blobConnectionInfo = mockk<BlobAccess.BlobContainerMetadata>()
+        )
+
+        val transformer = FhirTransformer(schema)
+        transformer.process(bundle)
+        assertThat(serviceRequest.note).hasSize(2)
+        @Suppress("ktlint:standard:max-line-length")
+        assertThat(serviceRequest.note)
+            .transform { it.map { note -> note.text } }
+            .containsOnly(
+                "OBX filtered for identifier=92142-9 - Influenza virus A RNA [Presence] in Respiratory specimen by NAA with probe detection;value = 260385009 - Negative OBX was removed due to your jurisdictional reporting rules indicating this result is not reportable.",
+                "OBX filtered for identifier=92141-1 -  Influenza virus B RNA [Presence] in Respiratory specimen by NAA with probe detection;value = 260385009 - Negative OBX was removed due to your jurisdictional reporting rules indicating this result is not reportable."
+            )
```

Comment:
Should there also be asserts for the extensions that are part of the `obs-to-note.yml`?

Reply:
Yep, can add it